src/test/java/com/google/enterprise/cloudsearch/fs/WindowsFileDelegateTest.java

```diff
@@ -105,45 +103,50 @@ public static void checkIfRunningOnWindows() {
   }
 
   private static class MockEventPusher implements RepositoryEventPusher {
-    Set<AsyncApiOperation> events = new HashSet<>();
+    Set<ApiOperation> events = new HashSet<>();
+    Map<ApiOperation, List<GenericJson>> responses = new HashMap<>();
+    Map<ApiOperation, Throwable> errors = new HashMap<>();
 
-    public void push(AsyncApiOperation event) {
-      events.add(event);
+    MockEventPusher respond(ApiOperation operation, GenericJson response) {
+      return respond(operation, Collections.singletonList(response));
     }
-  };
-
-  private static class SettableAsyncApiOperation extends AsyncApiOperation {
-    private GenericJson result;
-    private Throwable exception;
 
-    SettableAsyncApiOperation(ApiOperation operation, GenericJson result) {
-      super(operation);
-      this.result = result;
+    MockEventPusher respond(ApiOperation operation, List<GenericJson> response) {
+      responses.put(operation, response);
+      return this;
     }
 
-    SettableAsyncApiOperation(ApiOperation operation, Throwable throwable) {
-      super(operation);
-      this.exception = throwable;
+    MockEventPusher respond(ApiOperation operation, Throwable response) {
+      errors.put(operation, response);
+      return this;
     }
 
-    @Override
-    public ListenableFuture<List<GenericJson>> getResult() {
+    public ListenableFuture<List<GenericJson>> push(ApiOperation event) {
+      events.add(event);
       SettableFuture<List<GenericJson>> result = SettableFuture.create();
-      if (exception != null) {
-        result.setException(this.exception);
+      if (errors.get(event) != null) {
+        result.setException(errors.get(event));
       } else {
-        result.set(ImmutableList.of(this.result));
+        result.set(responses.get(event));
+      }
+      if (!(errors.containsKey(event) || responses.containsKey(event))) {
+        System.out.println("missing response for: " + event + "; returning " + result);
```

Comment:
Is this supposed to be here? Is there any action to take, or way to confirm whether this is expected or not?